[testing] Improve testing to include models, image inputs and autotuned parameters

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24 [WIP] Use PackedFunc to enable custom TVM operator registration from python
* #23 [benchmark] Add initial benchmark file
* #22 [compiler] Improve conversion from PyTorch IR to Relay
* **#21 [testing] Improve testing to include models, image inputs and autotuned parameters**
* #17 [compiler] Add None type handling with sentinel value
* #14 [compiler] Support operators with multiple outputs

bwasti committed:
[testing] Improve testing to include models, image inputs and autotuned parameters

wanchaol commented on test/util.py:
```diff
@@ -60,26 +61,33 @@ def f_impl(*args, **kwargs):
+
+          # jit the function and lower to TVM
+          torch_tvm.enable()
+          with autotvm.apply_history_best("test/autotvm_tuning.log"):
```
this actually will not be that transparent to the user. I think we might can make the `torch_tvm.enable` API take if we do autotvm and pass the log there, in the future we will need to think about how much tunable tvm api that we want to expose to the user.

bwasti replied:
everything is exposed to the user by default pretty much

wanchaol commented on test/test_models.py:
```diff
@@ -329,25 +329,119 @@ def resnext101_32x8d(pretrained=False, **kwargs):
+        torch.testing.assert_allclose(ref_out.topk(k).indices, tvm_out.topk(k).indices)
+
+    def test_resnet18(self):
+        self.model_test(resnet18)
```
curious, there should be somewhere import `resnet18` from torchvision, where i did not see those import statements though

bwasti replied:
it's defined above -- I just copied and pasted it in

bwasti committed:
Update on "[testing] Improve testing to include models, image inputs and autotuned parameters"

[testing] Improve testing to include models, image inputs and autotuned parameters

gh-metadata: pytorch tvm 21 gh/bwasti/9/head

bwasti committed:
Update on "[testing] Improve testing to include models, image inputs and autotuned parameters"

[testing] Improve testing to include models, image inputs and autotuned parameters

gh-metadata: pytorch tvm 21 gh/bwasti/9/head
